server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -197,6 +198,42 @@
     log.info("Created Appenderator for dataSource[%s].", schema.getDataSource());
   }
 
+  /**
+   * This constructor allows the caller to provide its own sink timeline and segment walker.
+   *
+   * It is used by UnifiedIndexerAppenderatorsManager which allows queries on data associated with multiple
+   * Appenderators.
+   */
+  AppenderatorImpl(
```

Comment:
can other constructor use this constructor instead of repeating everything ?

Reply:
Adjusted the constructors so one calls the other, the sink timeline handling in the constructor I added was also changed to accommodate that